### PR TITLE
[gh-release] initial implementation

### PR DESCRIPTION
**Pull request by dfuentes**
Github releases implementation.  Note that requires your project have a VERSION file with the version tag you want to create (with optional version description a la ark).

Tested here: https://circleci.com/gh/Clever/wag-test/46

### CODE REVIEW

File: circleci/github-release [hunk 1]
```diff
@@ -19,6 +17,42 @@ if [[ -z $ARTIFACTS_DIR ]]; then echo "Missing arg2 ARTIFACTS_DIR" && exit 1; fi
...
+: ${CIRCLE_PROJECT_USERNAME?"Missing required env var"}
+USER=$CIRCLE_PROJECT_USERNAME
 
-echo "Error! Not yet implemented"
-return 1
+if [ ! -e VERSION ]; then echo "Missing VERSION file" && exit 1; fi
+
+# Download github-release script
+echo "Downloading github-release tool"
+curl -sSL -o /tmp/github-release.tar.bz2 https://github.com/aktau/github-release/releases/download/v0.5.2/linux-amd64-github-release.tar.bz2
```
Comment (mohit) [inline circleci/github-release]: curl and untar can be done in one command. just a nit :p

File: circleci/github-release [hunk 2]
```diff
@@ -19,6 +17,42 @@ if [[ -z $ARTIFACTS_DIR ]]; then echo "Missing arg2 ARTIFACTS_DIR" && exit 1; fi
...
+  echo "Error creating release: $result"
+  exit 1
+fi
+
+for f in $ARTIFACTS_DIR; do
+    # treat directories and files differently
+    if [ -d $f ]; then
+        for ff in $(ls $f); do
+            echo -e "uploading $ff"
+            github-release upload -u $USER -r $REPO -t $TAG -n $ff -f $f/$ff
```
Comment (mohit) [inline circleci/github-release]: should we check if this fails?

File: circleci/github-release [hunk 3]
```diff
@@ -19,6 +17,42 @@ if [[ -z $ARTIFACTS_DIR ]]; then echo "Missing arg2 ARTIFACTS_DIR" && exit 1; fi
...
+        done
+    elif [ -f $f ]; then
+        echo -e "uploading $f"
+        github-release upload -u $USER -r $REPO -t $TAG -n $f -f $f
```
Comment (mohit) [inline circleci/github-release]: aain do we need to check i this fails?

File: circleci/github-release [hunk 4]
```diff
@@ -19,6 +17,42 @@ if [[ -z $ARTIFACTS_DIR ]]; then echo "Missing arg2 ARTIFACTS_DIR" && exit 1; fi
...
+    else
+        echo -e "$f is not a file or directory"
+        exit 1
+    fi
+done
```
Comment (mohit) [inline circleci/github-release]: We should proabably print out a summary of artifacts uploaded.

**Comment by mohit** [comment]
Small error checking questions.

File: circleci/github-release (hunk 2 above)
Comment (nathanleiby) [inline circleci/github-release]: Will the `set -e` above handle that for us?

Comment (nathanleiby) [inline circleci/github-release]: Ah, nvm. Looking above it appears to be returning a HTTP status code (422)

**Comment by nathanleiby** [comment]
👍

File: circleci/github-release (hunk 2 above)
Comment (dfuentes) [inline circleci/github-release]: A failure on this step is already pretty loud (as can be seen here: https://circleci.com/gh/Clever/wag-test/44), so I don't think we need specific error handling

File: circleci/github-release (hunk 3 above)
Comment (dfuentes) [inline circleci/github-release]: same answer as above

File: circleci/github-release (hunk 4 above)
Comment (dfuentes) [inline circleci/github-release]: Each artifact is echo'd as it is uploaded, so you do end up with a summary

**Pull request merged**